simulation/agent.py

```diff
@@ -223,15 +275,16 @@ def _update_vision(self, force=False) -> bool:
             return True
         return False
 
-    def tick(self, seen_agents: List['vision.AgentView'], noises: List['world.NoiseEvent']):
+    def tick(self, seen_agents: List['vision.AgentView']):
+        #Random entering and leaving towers -> Prove of concept. self.interact_with_tower(tower) lets the agent interact with a tower 
```

Comment:
Please don't put agent behaviour in the parent class, only do that in the implementation classes, and any behaviour should also go in the abstract methods (`on_tick`, `on_...`, ...), not the `agent.tick` method.